ocaml/lambda/printlambda.ml

```diff
@@ -208,26 +220,26 @@ let layout ppf layout_ = layout' true ppf layout_
 let return_kind ppf (mode, kind) =
   let smode = alloc_mode_if_local mode in
   match kind with
-  | Pvalue Pgenval when is_heap_mode mode -> ()
-  | Pvalue Pgenval -> fprintf ppf ": %s@ " smode
-  | Pvalue Pintval -> fprintf ppf ": int@ "
-  | Pvalue (Pboxedfloatval bf) ->
+  | Pvalue { raw_kind = Pgenval; _ } when is_heap_mode mode -> ()
+  | Pvalue { raw_kind = Pgenval; _ } -> fprintf ppf ": %s@ " smode
+  | Pvalue { raw_kind = Pintval; _ } -> fprintf ppf ": int@ "
+  | Pvalue { raw_kind = Pboxedfloatval bf; _ } ->
      fprintf ppf ": %s%s@ " smode (boxed_float_name bf)
-  | Pvalue (Parrayval elt_kind) ->
+  | Pvalue { raw_kind = Parrayval elt_kind; _ } ->
      fprintf ppf ": %s%sarray@ " smode (array_kind elt_kind)
-  | Pvalue (Pboxedintval bi) -> fprintf ppf ": %s%s@ " smode (boxed_integer_name bi)
-  | Pvalue (Pboxedvectorval (Pvec128 v)) ->
+  | Pvalue { raw_kind = Pboxedintval bi; _ } -> fprintf ppf ": %s%s@ " smode (boxed_integer_name bi)
+  | Pvalue { raw_kind = (Pboxedvectorval (Pvec128 v)); _ } ->
     fprintf ppf ": %s%s@ " smode (vec128_name v)
-  | Pvalue (Pvariant { consts; non_consts; }) ->
-    variant_kind value_kind' ppf ~consts ~non_consts
+  | Pvalue { raw_kind = (Pvariant { consts; non_consts; }); _ } ->
+    variant_kind (value_kind value_kind_non_null') ppf ~consts ~non_consts
```

Comment:
It feels wrong to exclude the `or_null` information when printing return kinds.

Reply:
This should be fixed with the new code.